proto/regen/data/v1alpha2/types.proto

```diff
@@ -0,0 +1,110 @@
+    // type specifies the IDType for the ID
+    IDType type = 1;
+
+    // hash represents the hash of the data based on the specified digest_algorithm and canonicalization algorithm
+    // for types other than ID_TYPE_RAW_UNSPECIFIED.
+    bytes hash = 2;
+
+    // digest_algorithm represents the hash digest algorithm.
+    DigestAlgorithm digest_algorithm = 3;
+
+    // media_type represents the MediaType for data with ID_TYPE_RAW_UNSPECIFIED. It should be left unset
+    // if type is not ID_TYPE_RAW_UNSPECIFIED.
+    MediaType media_type = 4;
+
+    // graph_canonicalization_algorithm represents the RDF graph canonicalization algorithm. It should
+    // be left unset if type is not ID_TYPE_GRAPH.
+    GraphCanonicalizationAlgorithm graph_canonicalization_algorithm = 5;
+}
+
+enum IDType {
+    // ID_TYPE_RAW_UNSPECIFIED specifies "raw" data which does not specify a deterministic, canonical encoding.
+    // Users of these hashes MUST maintain a copy of the hashed data which is preserved bit by bit.
+    // All other encodings in IDType specify a deterministic, canonical encoding allowing implementations to
+    // choose from a variety of alternative formats for transport and encoding while maintaining the guarantee
+    // that the canonical hash will not change. The media type for "raw" data used with ID_TYPE_RAW_UNSPECIFIED
+    // is defined by the MediaType enum.
+    ID_TYPE_RAW_UNSPECIFIED = 0;
+
+    // ID_TYPE_GRAPH specifies graph data that conforms to the RDF data model.
+    // The canonicalization algorithm used for an RDF graph is specified by GraphCanonicalizationAlgorithm.
+    ID_TYPE_GRAPH = 1;
+
+    // ID_TYPE_GEOGRAPHY specifies geographic data conforming to this canonicalization algorithm:
+    //
+    // We define the canonical lexical representation of a geography as its  [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) serialization with the following additional restrictions:
+    //
+    // ## **General Rules**
+    //
+    // - only the WGS 84 reference coordinate system is supported and all coordinates should be assumed to be in WGS 84
+    // - only 2D and 3D coordinates are supported, coordinates with linear referencing systems or an “M” component are unsupported
+    //
+    // ## **`POLYGON` Rules**
+    //
+    // - exterior linear rings must be defined in counterclockwise direction
+    // - interior linear rings must be defined in clockwise direction
+    // - where there is more than one ring, the first MUST be the exterior ring and other rings MUST be linear rings - this means that the area of a polygon MUST be continuous and unbroken. To represent multiple non-contiguous shapes, a `MULTIPOLYGON` should be used instead. Ex: in the case where inside of a hole in a polygon, there is an island that we want to include. Because the island is disjoint, this shape must be represented as a `MULTIPOLYGON` instead of a `POLYGON` with a hole and an island in the hole
```

Comment:
```suggestion
    // - where there is more than one ring, the first MUST be the exterior ring and other rings MUST be interior rings - this means that the area of a polygon MUST be continuous and unbroken. To represent multiple non-contiguous shapes, a `MULTIPOLYGON` should be used instead. Ex: in the case where inside of a hole in a polygon, there is an island that we want to include. Because the island is disjoint, this shape must be represented as a `MULTIPOLYGON` instead of a `POLYGON` with a hole and an island in the hole
```

I think this should this read that non exterior rings must be interior rings